src/main/java/com/amazon/opendistroforelasticsearch/security/securityconf/DynamicConfigFactory.java

```diff
@@ -81,7 +82,7 @@ private void loadStaticConfig() throws IOException {
     protected final Logger log = LogManager.getLogger(this.getClass());
     private final ConfigurationRepository cr;
     private final AtomicBoolean initialized = new AtomicBoolean();
-    private final List<DCFListener> listeners = new ArrayList<>();
+    private final EventBus eventBus = new EventBus("DynamicConfig");
```

Comment:
Wondering if this should be injected from the constructor as some other classes can post on the same event bus.

Reply:
Currently `eventBus` is not shared across `DynamicConfigFactory` class instances similar to how `listeners` was not shared. Till there is a use case where `eventBus` needs to be shared, I'd prefer to keep it encapsulated inside `DynamicConfigFactory`.